### PR TITLE
examples: replace the efl toolkit with sdl

### PR DESCRIPTION
**Pull request by hermet**
issue: https://github.com/thorvg/thorvg/issues/1244 https://github.com/thorvg/thorvg/issues/1991

### CODE REVIEW

**Comment by capnm** [comment]
Just thinking out loud: Could the source code possibly be split into "pure tvg*" and the part related to toolkits split into `./framework/sdl/` `./framework/glfw/` ... ?

**Comment by hermet** [comment]
> Just thinking out loud: Could the source code possibly be split into "pure tvg*" and the part related to toolkits split into `./framework/sdl/` `./framework/glfw/` ... ?

Maybe, if we apply the adapter or bridge, or something similar and if thorvg library has a windows layer over other frameworks dependency as you mentioned, definitely it should have like it.

However, the examples here are for illustration purposes only. I don't aim for a perfect design, as this is not a product or software intended for use. It is just a reference, serves as a direct guide on how to integrate the canvas into Windows and user system main loop. We don't need to introduce more complicated design and code for "our" maintanence, and thorvg never gonna introduce multiple frameworks for examples here. 

Anyway, in this improvement, we have made many enhancements to make it more intuitive and allow users to focus solely on the ThorVG practices. Please have a look at it and thanks your opinion.

File: examples/Capi.cpp [hunk 1]
```diff
@@ -272,115 +259,88 @@ void testCapi()
...
+    tvg_swcanvas_set_target(canvas, (uint32_t*)surface->pixels, surface->w, surface->pitch / 4, surface->h, TVG_COLORSPACE_ARGB8888);
+    tvg_swcanvas_set_mempool(canvas, TVG_MEMPOOL_POLICY_DEFAULT);
 
+    contents();
+
+    SDL_Event event;
+    auto running = true;
+    auto ptime = SDL_GetTicks();
+    auto elapsed = 0;
+    uint32_t tickCnt = 0;
```
Comment (mgrudzinska) [inline examples/Capi.cpp:297]: unused variable warning

File: examples/Example.h [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+    #ifdef _WIN32
+        auto rpath = _fullpath(buf, path, PATH_MAX);
+    #else
+        auto rpath = realpath(path, buf);
+    #endif
+
+        //open directory
+    #ifdef _WIN32
+        WIN32_FIND_DATA fd;
+        HANDLE h = FindFirstFileEx((path + "\\*").c_str(), FindExInfoBasic, &fd, FindExSearchNameMatch, NULL, 0);
```
Comment (SergeyLebedkin) [inline examples/Example.h:85]:
```
../examples/Example.h:85:42: error: invalid operands of types 'const char*' and 'const char [3]' to binary 'operator+'
   85 |         HANDLE h = FindFirstFileEx((path + "\\*").c_str(), FindExInfoBasic, &fd, FindExSearchNameMatch, NULL, 0);
      |                                     ~~~~ ^ ~~~~~
      |                                     |      |
      |                                     |      const char [3]
      |                                     const char*
```
Should be some thing like?
`HANDLE h = FindFirstFileEx((std::string(path) + "\\*").c_str(), FindExInfoBasic, &fd, FindExSearchNameMatch, NULL, 0);`

File: examples/CustomTransform.cpp [hunk 3]
```diff
@@ -20,203 +20,93 @@
...
+
+        //scale y
+        m.e22 = 1 + (progress * 2.0f);
+
+        //rotation
+        constexpr auto PI = 3.141592f;
+        auto degree = 45.0f;
+        auto radian = degree / 180.0f * PI;
+        auto cosVal = cosf(radian);
+        auto sinVal = sinf(radian);
```
Comment (SergeyLebedkin) [inline examples/CustomTransform.cpp:76]: Windows: missed `#include <cmath>`

**Comment by capnm** [comment]
> ... Anyway, in this improvement, we have made many enhancements to make it more intuitive and allow users to focus solely on the ThorVG practices. Please have a look at it and thanks your opinion.

Would sdl also work with the wg_engine? I have only tried the glfw successfully. Yes, this has to serve as a quick demonstration of how it works for "someone" new to tvg. We did some AI lessons recently and my experiment with the current AI tools questioned for a simple ThorVG example resulted in completely incorrect and unusable code, thus showing up as an AI failure. Probably the API docs? are unreadable for the robots ;-)

**Comment by hermet** [comment]
> Would sdl also work with the wg_engine?

Yes, please check this guide: https://github.com/thorvg/thorvg/wiki/WebGPU-Raster-Engine-Development

File: examples/Example.h [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+#include <cmath>
+#include <vector>
+#include <fstream>
+#include <iostream>
+#include <thread>
+#include <thorvg.h>
+#include <SDL.h>
+#include <SDL_syswm.h>
+#ifdef _WIN32
+    #define WIN32_LEAN_AND_MEAN
```
Comment (SergeyLebedkin) [inline examples/Example.h:33]: WIN32_LEAN_AND_MEAN redefined in config.h

**Pull request merged**